### PR TITLE
[A0-1141] Enforce disabled pruning

### PR DESCRIPTION
**Pull request by h4nsu**
# Description

Quick and a bit dirty fix for the fact that we don't support pruning. The value of the corresponding flag is forcefully set at the beginning of `main()`.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: bin/node/src/main.rs [hunk 1]
```diff
@@ -9,7 +9,20 @@ use sc_network::config::Role;
 use sc_service::PartialComponents;
 
 fn main() -> sc_cli::Result<()> {
-    let cli = Cli::parse();
+    let mut cli = Cli::parse();
+
+    match cli.run.import_params.pruning_params.pruning.as_ref() {
+        Some(mode) => {
+            if mode != "archive" {
+                println!("Pruning not supported. Switching to 'archive' mode.");
```
Comment (DamianStraszak) [inline bin/node/src/main.rs:17]: It's better to log it I think -- `info!` probably.

**Review by DamianStraszak — APPROVED** [review]

File: bin/node/src/main.rs (hunk 1 above)
Comment (maciejnems) [reply bin/node/src/main.rs:17]: Logger is not initialized here yet :( Actually, where is it initialized @h4nsu? Haven't found that. Does it happen inside Substrate?

Comment (h4nsu) [reply bin/node/src/main.rs:17]: It's not possible to use the logger here, it gets initialized later, in differerent `match` branches separately.

**Review by maciejnems — APPROVED** [review]

File: bin/node/src/main.rs (hunk 1 above)
Comment (h4nsu) [reply bin/node/src/main.rs:17]: @maciejnems I think somewhere in `cli.create_runner`

Comment (h4nsu) [reply bin/node/src/main.rs:17]: [client/cli/src/config.rs](https://github.com/Cardinal-Cryptography/substrate/blob/5f0aa1feb7250ac7b8c1b9928f87b2420b530e22/client/cli/src/config.rs#L651)

File: bin/node/src/main.rs [hunk 2]
```diff
@@ -9,7 +9,20 @@ use sc_network::config::Role;
...
+                cli.run.import_params.pruning_params.pruning = Some(String::from("archive"));
+            }
+        }
+        None => {
+            cli.run.import_params.pruning_params.pruning = Some(String::from("archive"));
+        }
+    }
```
[the remark this answers is not in the document]
Comment (h4nsu) [reply bin/node/src/main.rs:24]: I don't know :/ I find it kinda cryptic and difficult to read, but maybe I'm just not familiar with idiomatic Rust ^^' I changed it according to your suggestion.

[the remark this answers is not in the document]
Comment (maciejnems) [reply bin/node/src/main.rs]: Could be even shorter :3
```
if cli
    .run
    .import_params
    .pruning_params
    .pruning
    .replace(String::from("archive"))
    .map_or(false, |x| x != "archive")
{
    println!("Pruning not supported. Switching to 'archive' mode.");
}
```

[the remark this answers is not in the document]
Comment (h4nsu) [reply bin/node/src/main.rs]: Even more cryptic ^^

**Pull request merged**